modules/internal/ExportWrappers.chpl

```diff
@@ -23,65 +23,52 @@ module ExportWrappers {
   private use SysCTypes;
 
   // Actual definition is in "runtime/include/chpl-export-wrappers.h".
-  extern record chpl_bytes_wrapper {
+  extern record chpl_byte_buffer {
     var isOwned: int(8);
     var data: c_ptr(c_char);
     var size: uint(64);
   }
 
-  // May need to call this in one of the conversion routines.
-  extern proc chpl_bytes_wrapper_free(cb: chpl_bytes_wrapper);
+  extern proc chpl_byte_buffer_free(cb: chpl_byte_buffer);
 
   //
   // TODO: Using type aliases to resolve a type shouldn't be necessary. The
   // compiler should be able to figure this out on its own.
   //
-  type chpl__exportTypeCharPtr = c_ptr(c_char);
-  type chpl__exportTypeChplBytesWrapper = chpl_bytes_wrapper;
+  type chpl__exportTypeChplBytesWrapper = chpl_byte_buffer;
 
-  private proc chpl__exportCopyStringBuffer(s: string): c_ptr(c_char) {
-    const nBytes = s.numBytes;
-    const src = s.c_str():c_void_ptr;
-    var result = c_malloc(c_char, nBytes + 1):c_ptr(c_char);
-    c_memcpy(result, src, nBytes);
-    result[nBytes] = 0;
+  // Generic, but both string and bytes have the same implementation.
+  proc chpl__exportRetStringOrBytes(ref val): chpl_byte_buffer {
+    var result: chpl_byte_buffer;
+    result.isOwned = val.isowned:int(8);
+    result.data = val.buff:c_ptr(c_char);
+    result.size = val.size:uint(64);
+    // Assume ownership of the string/bytes record's internal buffer.
+    val.isowned = false;
+    result.size = val.size:uint(64);
```

Comment:
You change the size twice here, I don't think that's necessary?

Reply:
Good catch, thanks!